.github/workflows/e2e-tests.yml

```
@@ -39,8 +39,39 @@ jobs:
         id: e2e-matrix
         uses: ./.github/actions/build-e2e-matrix
 
+  # if this is a test on a release branch, we need to check the build requirements
+  get-build-requirements:
+    if: |
+      !cancelled() &&
+      contains(github.ref, 'release-x')
```

```suggestion
      startsWith(github.ref, 'release-x')
```

Maybe it's safer to use https://docs.github.com/en/actions/learn-github-actions/expressions#startswith?

---

it would have to be `startsWith('refs/heads/release-x')` and i'd rather be more permissive here. if it hits on a weirdly named branch the worst that happens is it wastes 12 seconds checking the build version requirements

---

`github.ref_name` gives you just the branch name

either way is fine with me